blog/tidy-finance-dummy-data/index.qmd

```diff
@@ -0,0 +1,422 @@
+        mutate(
+          exchcd = case_when(
+            exchange == "NYSE" ~ sample(c(1, 31), n()),
+            exchange == "AMEX" ~ sample(c(2, 32), n()),
+            exchange == "NASDAQ" ~ sample(c(3, 33), n())
+          ),
+          siccd = case_when(
+            industry == "Agriculture" ~ sample(1:999, n()),
+            industry == "Mining" ~ sample(1000:1499, n()),
+            industry == "Construction" ~ sample(1500:1799, n()),
+            industry == "Manufacturing" ~ sample(1800:3999, n()),
+            industry == "Transportation" ~ sample(4000:4899, n()),
+            industry == "Utilities" ~ sample(4900:4999, n()),
+            industry == "Wholesale" ~ sample(5000:5199, n()),
+            industry == "Retail" ~ sample(5200:5999, n()),
+            industry == "Finance" ~ sample(6000:6799, n()),
+            industry == "Services" ~ sample(7000:8999, n()),
+            industry == "Public" ~ sample(9000:9999, n())
+          )
+        )
+    }
+  )
+```
+
+Next, we construct three panels of stock data with varying frequencies: yearly, monthly, and daily. We begin by creating the `stock_panel_yearly` panel. To achieve this, we combine the `stock_identifiers` table with a new table containing the variable `year` from `time_series_years`. The `crossing()` function ensures that we get all possible combinations of the two tables. After combining, we select only the `gvkey` and `year` columns for our final yearly panel.
+
+Next, we construct the `stock_panel_monthly` panel. Similar to the yearly panel, we use the `crossing()` function to combine `stock_identifiers` with a new table that has the `month` variable from `time_series_months`. After merging, we select the columns `permno`, `gvkey`, `month`, `siccd`, `industry`, `exchcd`, and `exchange` to form our monthly panel.
+
+Lastly, we create the `stock_panel_daily` panel. We combine `stock_identifiers` with a table containing the `date` variable from `time_series_days`. After merging, we retain only the `permno` and `date` columns for our daily panel.
+
+```{r}
+stock_panel_yearly <- crossing(
+  stock_identifiers, 
+  tibble(year = time_series_years)
+) |> 
+  select(gvkey, year)
+
+stock_panel_monthly <- crossing(
+  stock_identifiers, 
+  tibble(month = time_series_months)
+) |> 
+  select(permno, gvkey, month, siccd, industry, exchcd, exchange)
+
+stock_panel_daily <- crossing(
+  stock_identifiers, 
+  tibble(date = time_series_days)
+)|> 
+  select(permno, date)
+```
+
+### Dummy `beta` table
+
+We then proceed to simulate beta values for our `stock_panel_monthly` table. We generate simulated monthly beta values `beta_monthly` using the `rnorm()` function with a mean and standard deviation of 1. For daily beta values `beta_daily`, we take the simulated monthly beta and add a small random noise to it. This noise is generated again using the `rnorm()` function, but this time we divide the random values by 100 to ensure they are small deviations from the monthly beta.
```

Comment:
Consider replacing "simulate" in this paragraph.

Reply:
Removed all occurrences of "simualted" or "simualte"